neqo-http3/src/connection.rs

```diff
@@ -79,6 +92,193 @@ impl Http3State {
     }
 }
 
+/**
+# HTTP/3 core implementation
+
+This is the core implementation of HTTP/3 protocol. It implements most of the features of the
+protocol. Http3Client and Http3ServerHandler implement only client and server side bbehavior.
+
+The API consists of:
+- functions that correspond to the Http3Client Http3ServerHandler API:
+  - `new`
+  - `close`
+  - `fetch` -  only used by the client-side implementation
+  - `read_data`
+  - `stream_reset_send`
+  - `stream_stop_sending`
+  - `cancel_fetch`
+  - `stream_close_send`
+  - `webtransport_create_session` -  only used by the client-side implementation
+  - `webtransport_session_accept` -  only used by the server-side implementation
+  - `webtransport_close_session`
+  - `webtransport_create_stream_local` -  this function is called when an application wants to open
+     a new WebTransport stream. For example `Http3Client::webtransport_create_stream` will call
+     this function.
+  - `webtransport_create_stream_remote` -  this is called when a WebTransport stream has been
+     opened by the peer and this function sets up the appropriate handler for the stream.
+- functions that are called by `process_http3`
+  - `process_sending` - some send-streams are buffered streams(see the Streams section) and this
+     function is called to trigger sending of the buffer data.
+- functions that are called to  handle `ConnectionEvent`s:
+  - `add_new_stream`
+  - `handle_stream_readable`
+  - `handle_stream_reset`
+  - `handle_stream_stop_sending`
+  - `handle_state_change`
+  - `handle_zero_rtt_rejected`
+- Additional functions:
+  - `set_features_listener`
+  - `stream_has_pending_data`
+  - `has_data_to_send`
+  - `add_streams`
+  - `add_recv_stream`
+  - `queue_control_frame`
+  - `queue_update_priority`
+  - `set_0rtt_settings`
+  - `get_settings`
+  - `state`
+  - `webtransport_enabled`
+
+## Streams
+
+Http3Connection holds a list of stream handlers. Each send and receive-handler is registered in
+`send_streams` and `recv_streams`. Unidirectional streams are registered only on one of the lists
+and bidirectional streams are registered in both lists and the 2 handlers are independent, e.g. one
+can be closed and removed ane second may still be active.
+
+The only streams that are not registered are the local control stream, local QPACK decoder stream,
+and local QPACK encoder stream. These streams are send-streams and sending data on this stream is
+handled a bit differently. This is done in the `process_sending` function, i.e. the control data is
+sent first and QPACK data is sent after regular stream data is sent because this stream may have
+new data only after regular streams are handled (TODO we may improve this a bit to send QPACK
+commands before headers.)
+
+There are the following types of streams:
```

Comment:
Are these types of streams or the types of the new stream events?

Reply:
They are types of streams. 
For example calling the WebTransportSession::stream_type() function will return  Http3StreamType::ExtendedConnect.

Http3Connection needs to handle some stream types differently therefore it needs to know the type of receive- and send-stream-handlers.